### PR TITLE
24658 Implement Staff Search by Filing ID

### PR DESCRIPTION
**Pull request by meawong**
*Issue:*https://github.com/bcgov/entity/issues/24658 

*Description of changes:*
- Added new filings search API call to fetch filing
- Extend search functionality to accept and handle filingID
- As per Sev, the code to handle auto-expanding the file does not exist yet in the new Business Dashboard but it will expect the filingID which we are passing as a param in the URL

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-3186-89fgsbuz.web.app

**Comment by meawong** [comment]
> Temporary Url for review: https://bcregistry-account-dev--pr-3186-89fgsbuz.web.app

Business Identifier Test: BC0883798
Filing Number Test: 154799

File: auth-web/src/views/auth/staff/IncorporationSearchResultView.vue
Comment (severinbeauvais) [inline auth-web/src/views/auth/staff/IncorporationSearchResultView.vue:1]: I was hoping the filing search result would show what type of filing this is, but I didn't look at the UI design. What do you think? Can you ask Jacqueline?

image

File: auth-web/src/stores/business.ts [hunk 1]
```diff
@@ -268,6 +269,19 @@ export const useBusinessStore = defineStore('business', () => {
...
   }
 
+  async function loadFiling () {
+    const filingID = ConfigHelper.getFromSession(SessionStorageKeys.FilingIdentifierKey)
+    state.filingID = filingID
+    const response = await BusinessService.searchFiling(filingID).catch(() => null)
+    if (response?.status === 200) {
+      ConfigHelper.addToSession(SessionStorageKeys.BusinessIdentifierKey, response?.data.filing.business.identifier)
+    } else if (response?.status === 404) {
+      throw Error('No match found for Filing Number')
```
Comment (severinbeauvais) [inline auth-web/src/stores/business.ts:279]: I'm not seeing this error if I search for a non-existent filing id.

File: auth-web/src/views/auth/staff/IncorporationSearchResultView.vue
Comment (meawong) [reply auth-web/src/views/auth/staff/IncorporationSearchResultView.vue:1]: i will get Jacqueline's input!

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-3186-89fgsbuz.web.app

File: auth-web/src/stores/business.ts (hunk 1 above)
Comment (severinbeauvais) [reply auth-web/src/stores/business.ts:286]: Looks good now. Thanks!

image

File: auth-web/src/views/auth/staff/IncorporationSearchResultView.vue [hunk 2]
```diff
@@ -176,8 +177,17 @@ export default class IncorporationSearchResultView extends Vue {
...
-  async entityDashboardEvent () {
-    window.location.href = `${ConfigHelper.getBusinessURL()}${this.currentBusiness.businessIdentifier}`
+  async manageBusinessEvent () {
+    const businessIdentifier = this.currentBusiness.businessIdentifier
+    const filingId = this.filingID
+
+    let url = `${ConfigHelper.getBusinessURL()}${businessIdentifier}`
+
+    if (filingId) {
+      url += `?filingId=${filingId}`
```
Comment (severinbeauvais) [inline auth-web/src/views/auth/staff/IncorporationSearchResultView.vue:187]: This should actually be `filing_id=`.

Example: `https://dev.business.bcregistry.gov.bc.ca/BC0884782?filing_id=155306`

**Review by severinbeauvais — APPROVED** [review]
I am pre-approving this PR as I trust you to follow up on the last couple of things.

File: auth-web/src/views/auth/staff/IncorporationSearchResultView.vue
Comment (meawong) [reply auth-web/src/views/auth/staff/IncorporationSearchResultView.vue:1]: Jacqueline and Mihai have confirmed that we should leave it as is

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-3186-89fgsbuz.web.app

**Comment by severinbeauvais** [comment]
@seeker25 OK if we merge this change?

Also, there's another change coming soon for [24920](https://app.zenhub.com/workspaces/entities---olga-65af15f59e89f5043c2911f7/issues/gh/bcgov/entity/24920) to facilitate cutting over from old MBR to new BRD.

For both of these, is there a path forward (ie, release) to go to Test and then Prod?

**Comment by seeker25** [comment]
@severinbeauvais yes, ideally not into PROD until the new year though - you should be able to both push to TEST whenever you're ready

**Pull request merged**